Implement CurrentDateUpdater and handle date-to-string representation

WooKaiNing opened the pull request.

WooKaiNing committed:
Modify exception handling in Expense classes

WooKaiNing committed:
Add JUnit testing for Expense classes

WooKaiNing committed:
Fix checkstyle errors

WooKaiNing committed:
Fix checkstyle errors

WooKaiNing committed:
Merge branch 'master' into expense

WooKaiNing committed:
Implement JUnit test for ExpenseStatement

WooKaiNing committed:
Implement Expense summation and JUnit testing for ExpenseStatement

WooKaiNing committed:
Modify JUnit testing

WooKaiNing committed:
Fix Checkstyle error

WooKaiNing committed:
Merge branch 'master' into Expense

WooKaiNing committed:
Fix Checkstyle error

WooKaiNing committed:
Merge branch 'master' into Expense

WooKaiNing committed:
Merge branch 'Expense' of https://github.com/WooKaiNing/tp into Expense

WooKaiNing committed:
Modify viewExpenses output in EXPECTED.TXT

mcmc101001 reviewed (changes requested):
I think we should fix merge conflicts first

WooKaiNing committed:
Merge branch 'master' into Expense

WooKaiNing committed:
Add JavaDoc to classes and methods

mcmc101001 reviewed (approved):
LGTM

WooKaiNing committed:
Implement printing of date and modify JUnit testing

WooKaiNing committed:
Implement CurrentDate class

WooKaiNing committed:
Modify CurrentDate class

WooKaiNing committed:
Modify input.txt and EXPECTED.TXT

WooKaiNing committed:
Revert input.txt/EXPECTED.TXT and modify runtest.bat

WooKaiNing committed:
Modify runtest.bat

WooKaiNing committed:
Modify runtest.bat

WooKaiNing committed:
Modify runtest.bat

WooKaiNing committed:
Modify runtest.bat

WooKaiNing committed:
Revert runtest.bat and modify input.txt/EXPECTED.TXT

WooKaiNing commented:
Modified input.txt to only include expense/income that have a fixed date for now, as dynamic updating of EXPECTED.TXT is required for expense/income without date input -> have implemented a CurrentDate class that will write to the EXPECTED.TXT file to update placeholder to current date, but this only works on macos and ubuntu after updating runtest.sh().

not sure if its bc i didn't modify runtest.bat correctly, thus doesnt work for windows

WooKaiNing committed:
Modify CurrentDateUpdater class and text-ui-test

WooKaiNing committed:
Modify runtest.bat

WooKaiNing committed:
Modify CurrentDateUpdater

WooKaiNing committed:
Fix Checkstyle issues

WooKaiNing committed:
Modify text-ui-test

WooKaiNing commented:
Implemented CurrentDateUpdater class to dynamically update EXPECTED.TXT file with current date (for input in INPUT.TXT where date is not specified). However, this somehow only works for Java CI / build (ubuntu-latest) (pull_request) and Java CI / build (macos-latest) (pull_request).

Hence, modified input.txt to only include expense/income input with specified date for now, will need to further refine code such that runtest.bat() works too.

danielpappa commented on src/main/java/cashleh/CurrentDateUpdater.java:
Great thinking and nice implementation.

danielpappa commented on src/main/java/cashleh/CurrentDateUpdater.java:
```diff
@@ -0,0 +1,41 @@
+        String formattedDate = currentDate.toString(); // Customize the date format as needed
+
+        // Replace all instances of the placeholder with the current date
+        List<String> updatedLines = lines.stream()
```
Great coding!

WooKaiNing commented:
Fixes #45
